hookbook.sh

```diff
@@ -108,36 +110,35 @@ case "${__hookbook_shellname}" in
     # accomplishing is to route as much tracing output as possible to
     # /dev/null, whilst keeping any stderr output generated by the handler on
     # stderr.
-    trap '__hookbook_debug_trap "$_"' DEBUG
+    \trap '__hookbook_debug_trap "$_"' DEBUG
 
     hookbook_add_hook() {
-      local fn=$1
+      \local fn=$1
 
-      if [[ ! "${PROMPT_COMMAND}" == *" $fn "* ]]; then
+      [[ ! "${PROMPT_COMMAND}" == *" $fn "* ]] && {
         # This is essentially:
         #   PROMPT_COMMAND="${fn}; ${PROMPT_COMMAND}"
         # ...except with weird magic to toggle off `-x` if it's set, much like
         # in the DEBUG trap above.
         PROMPT_COMMAND="{
-          if [[ \$- =~ x ]];
-          then set +x; ${fn} precmd 2>&3; set -x;
-          else ${fn} precmd 2>&3;
-          fi;
+          [[ \$- =~ x ]] && {
+            set +x; ${fn} precmd 2>&3; set -x;
+          } || {
+            ${fn} precmd 2>&3;
+          }
         } 4>&2 2>/dev/null 3>&4;
         ${PROMPT_COMMAND}"
-      fi
+      }
 
       __hookbook_array_contains "${fn}" "${__hookbook_functions[@]}" \
         || __hookbook_functions+=("${fn}")
     }
-    ;;
-  *)
-    >&2 \echo "hookbook is not compatible with your shell (${__hookbook_shell})"
-    \return 1
-    ;;
-esac
-
-unset __hookbook_shell
-unset __hookbook_shellname
+}
+
+[[ "${__hookbook_shellname}" == "zsh" ]] && __hookbook_setup_zsh && \unset __hookbook_shell __hookbook_shellname && \return 0
```

Comment:
Kind of too bad that this creates two more toplevel names. Could we instead do something like:

```
[[ "${__hookbook_shellname}" == "zsh" ]] && {

}
[[ "${__hookbook_shellname}" == "bash" ]] && {

}
\unset __hookbook_shell __hookbook_shellname
```

Reply:
I did this change before discovering that pattern.  Thanks for the catch.